src/workerd/jsg/resource.h

```diff
@@ -597,6 +597,163 @@ class DynamicResourceTypeMap {
   friend class ObjectWrapper;
 };
 
+// ======================================================================================
+// NamedIntercept implementation
+
+class JsValue;
+
+// A utility class that allows dynamic evaluation of properties on the JavaScript wrapper.
+// TODO(soon): Currently, this mechanism only supports read-only properties. The setter
+// interceptor is not implemented. That is, for a known dynamic name like `foo`, it is not
+// yet possible to set `obj.foo = whatever` in JS. However, it remains possible to
+// set unknown properties on the object (properties for which `queryNamed()` and `getNamed()`
+// would return kj::none).
+//
+// struct ProxyImpl: public jsg::Object,
+//                   public jsg::NamedIntercept {
```

Comment:
@jasnell It surprises me that this is based on inheritance, and the use of multiple inheritance makes me nervous since if you don't declare `jsg::Object` as the first superclass, stuff will break.

Is this `NamedIntercept` base class actually necessary? It looks like the implementation is templated such that the calls are always made on the derived class anyway...

Reply:
I actually ran into this problem the other day, declared the `jsg::Object` (in my case, `Fetcher`) after `NamedIntercept` and got segfaults for a while.

Reply:
> Is this `NamedIntercept` base class actually necessary?

Probably not. I put this together fairly quickly so that @MellowYarker could make progress, fully expecting that it could be refined. I think we ought to be able to drop the base class and I want to see if I can make the three methods that need to be implemented here a bit more ergonomic.